### PR TITLE
Feature/modal

### PR DESCRIPTION
**Pull request by toriroe**
## Describe your changes

Created reusable modal component using Headless UI's 'Dialog' component and animations. Also Modal context and useModal hook to access 'isOpen, currentModal, openModal, closeModal' throughout app

## Issue ticket number and link

Drop-11
https://tripleten-apiary.atlassian.net/jira/software/projects/DROP/boards/5?selectedIssue=DROP-11

## Checklist before requesting a review

- [X ] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ X] I have made any necessary updates to Storybook to accompany these changes
- [ X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by cgenardini — COMMENTED** [review]
I believe the buttonIcon was removed from the main branch.

import { ButtonIcon } from "../button/button";

**Comment by toriroe** [comment]
Thanks, I moved the task to the 'blocked' column for the time being

File: src/components/modal/modal.tsx [hunk 1]
```diff
@@ -35,18 +37,24 @@ export const Modal = ({ title, children }: modalProps) => {
...
           <div className="flex justify-between z-10">
             <Dialog.Title className="font-text text-section-subtext font-medium pb-8">
               {title}
             </Dialog.Title>
-            <ButtonIcon
+            <Button
+              variant="icon"
               onClick={() => setIsOpen(false)}
-              icon={exitButton}
+              icon={faX as IconProp}
```
Comment (cgenardini) [inline src/components/modal/modal.tsx:48]: Try  icon={faX as IconDefinition} to match the button component type.

File: src/components/modal/modal.tsx [hunk 2]
```diff
@@ -1,15 +1,17 @@
 import { ReactNode, useState } from "react";
 import { Dialog, Transition } from "@headlessui/react";
-import exitButton from "../../images/exit-modal-button-min.svg";
-import { ButtonIcon } from "../button/button";
+import { Button } from "../button/button";
+import { faX } from "@fortawesome/free-solid-svg-icons";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
```
Comment (cgenardini) [inline src/components/modal/modal.tsx:5]: import { IconDefinition } from "@fortawesome/fontawesome-common-types";
Try switching to IconDefinition instead to match type and replacing the IconProp. I switched it and the error went away.

**Review by cgenardini — COMMENTED** [review]
Try switching the IconProp to match the button component. I left a couple comments. Hope this is helpful!

File: src/components/modal/modal.tsx (hunk 2 above)
Comment (toriroe) [reply src/components/modal/modal.tsx]: I made the changes, and this did resolve the error. Thanks!

**Review by cgenardini — COMMENTED** [review]
This is still getting an error from the button.stories

**Pull request closed without merging**